deepchecks/nlp/text_data.py

```diff
@@ -0,0 +1,488 @@
+
+    def _validate_and_set_label(self, label: t.Optional[TTextLabel]):
+        """Validate and process label to accepted formats."""
+        # If label is not set, create an empty label of nulls
+        if label is None:
+            self._has_label, self._label = False, [None] * len(self._text)
+            return
+
+        self._has_label = True
+        if not is_sequence_not_str(label):
+            raise DeepchecksValueError('label must be a Sequence')
+
+        if not len(label) == len(self._text):
+            raise DeepchecksValueError('label must be the same length as raw_text')
+
+        if self.task_type == TaskType.TEXT_CLASSIFICATION:
+            if all((isinstance(x, collections.abc.Sequence) and not isinstance(x, str)) for x in label):
+                self._is_multilabel = True
+                multilabel_error = 'multilabel was identified. It must be a Sequence of Sequences of 0 or 1.'
+                if not all(all(y in (0, 1) for y in x) for x in label):
+                    raise DeepchecksValueError(multilabel_error)
+                if any(len(label[0]) != len(label[i]) for i in range(len(label))):
+                    raise DeepchecksValueError('All multilabel entries must be of the same length, which is the number'
+                                               ' of classes.')
+            elif not all(isinstance(x, (str, int)) for x in label):
+                raise DeepchecksValueError('label must be a Sequence of strings or ints or a Sequence of Sequences'
+                                           'of strings or ints (for multilabel classification)')
+
+        if self.task_type == TaskType.TOKEN_CLASSIFICATION:
+            token_class_error = 'label must be a Sequence of Sequences of either strings or integers'
+            if not all(isinstance(x, collections.abc.Sequence) for x in label):
+                raise DeepchecksValueError(token_class_error)
+
+            for i in range(len(label)):  # TODO: Runs on all labels, very costly
+                if not (all(isinstance(x, str) for x in label[i]) or all(isinstance(x, int) for x in label[i])):
+                    raise DeepchecksValueError(token_class_error)
+                if not len(label[i]) == len(self._tokenized_text[i]):
+                    raise DeepchecksValueError(f'label must be the same length as tokenized_text. '
+                                               f'However, for sample index {self.index[i]} of length '
+                                               f'{len(self._tokenized_text[i])} received label of '
+                                               f'length {len(label[i])}')
+        self._label = list(label)
+
+    def copy(self: TDataset, rows_to_use: t.Optional[t.Sequence[t.Any]] = None) -> TDataset:
+        """Create a copy of this Dataset with new data."""
+        cls = type(self)
+        get_logger().disabled = True  # Make sure we won't get the warning for setting class in the non multilabel case
```

Comment:
better to save state because maybe user has already disbaled it before and then you gonna enable it